### PR TITLE
Enable soft opt-ins for single contributors

### PR DESCRIPTION
**Pull request by jonflynng**
## Why are we doing this?

We want to enable a customer's soft opt-ins when they make a single contribution so we can keep in touch with them and send them personalised emails.

## Changes

- Adds a new service in the `services` package of the support-payments-api to initialise an SQS client and send a message to the soft opt-ins queue
- Sends the message in parallel alongside other jobs in the PaymentBackend.scala file in [this line here](https://github.com/guardian/support-frontend/blob/main/support-payment-api/src/main/scala/backend/PaymentBackend.scala#L65](https://github.com/guardian/support-frontend/blob/main/support-payment-api/src/main/scala/backend/PaymentBackend.scala#L65) )
- Updates unit tests
- Add queue permissions to the CDK cloudformation

## to-do before merging
 - [x] Test in CODE
 - [x] Add feature flag
 - [x] Add feature switch JSON file to dev, code and prod

### CODE REVIEW

File: support-payment-api/src/main/scala/services/SoftOptInsService.scala [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+class SoftOptInsService(sqsClient: AmazonSQSAsync, queueUrlResponse: Future[Either[SoftOptInsServiceError, String]])
+    extends StrictLogging {
+  def sendMessage(
+      message: Message,
+  )(implicit executionContext: ExecutionContext): EitherT[Future, SoftOptInsServiceError, Unit] = {
+    logger.info(s"Preparing to send message: ${message.asJson.noSpaces}")
+
+    message.identityId match {
+      case None =>
+        val error = SoftOptInsServiceError("Identity ID is None, soft opt-ins cannot be set")
```
Comment (jonflynng) [inline support-payment-api/src/main/scala/services/SoftOptInsService.scala:27]: @rupertbates  Will this happen often?

File: cdk/lib/payment-api.ts [hunk 2]
```diff
@@ -131,13 +131,20 @@ export class PaymentApi extends GuStack {
...
             resources:
               this.stage === "PROD"
                 ? [emailSqsCodeArn.valueAsString, emailSqsProdArn.valueAsString]
                 : [emailSqsCodeArn.valueAsString],
           }),
+          new GuAllowPolicy(this, "SoftOptInsSqsMessages", {
+            actions: ["sqs:GetQueueUrl", "sqs:SendMessage"],
+            resources:
+              this.stage === "PROD"
+                ? [`arn:aws:sqs:${this.region}:${this.account}:soft-opt-in-consent-setter-queue-PROD`,]
```
Comment (jonflynng) [inline cdk/lib/payment-api.ts:145]: is it better to import an exported ARN instead?

File: support-payment-api/src/main/scala/services/SoftOptInsService.scala (hunk 1 above)
Comment (rupertbates) [reply support-payment-api/src/main/scala/services/SoftOptInsService.scala:27]: I don't think it should ever happen

File: cdk/lib/payment-api.ts (hunk 2 above)
Comment (michaelbjacobson) [reply cdk/lib/payment-api.ts:145]: I don't think there's anything wrong with this, and I _think_ it's fairly idiomatic GuCDK use.

**Review by michaelbjacobson — APPROVED** [review]
Looks good to me! 👍🏼

File: support-payment-api/src/main/scala/backend/AmazonPayBackend.scala [hunk 3]
```diff
@@ -47,7 +38,8 @@ class AmazonPayBackend(
     val acquisitionsStreamService: AcquisitionsStreamService,
     val databaseService: ContributionsStoreService,
     val supporterProductDataService: SupporterProductDataService,
-    switchService: SwitchService,
```
Comment (jonflynng) [inline support-payment-api/src/main/scala/backend/AmazonPayBackend.scala:50]: this line without `val` was giving compiler errors?

**Review by rupertbates — APPROVED** [review]
Looks good 👍

**Pull request merged**